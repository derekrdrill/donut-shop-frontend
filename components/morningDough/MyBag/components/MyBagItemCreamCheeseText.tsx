import * as React from 'react';
import { Typography } from '@mui/material';

import { MyBagItem } from '../../../../context/types/MyBagItem';

type MyBagItemCreamCheeseTextProps = {
  myBagItem: MyBagItem;
};

const MyBagItemCreamCheeseText = ({ myBagItem }: MyBagItemCreamCheeseTextProps) =>
  myBagItem.subCategory === 'bagels' && myBagItem.creamCheese ? (
    <Typography data-testid='cream-cheese-text' component='p' variant='caption'>
      {`${myBagItem.creamCheese?.replace(
        /([A-Z]+)*([A-Z][a-z])/g,
        '$1 $2',
      )} CREAM CHEESE`.toUpperCase()}
    </Typography>
  ) : (
    <React.Fragment />
  );

export default MyBagItemCreamCheeseText;
