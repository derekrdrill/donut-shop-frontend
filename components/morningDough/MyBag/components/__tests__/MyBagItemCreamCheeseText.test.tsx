import '@testing-library/jest-dom';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import MyBagItemCreamCheeseText from '../MyBagItemCreamCheeseText';

describe('MyBagItemCreamCheeseText unit tests', () => {
  it('renders as expected', () => {
    const result = renderer
      .create(
        <MyBagItemCreamCheeseText
          myBagItem={{
            orderID: '1',
            menuItemID: 'plainBagel',
            name: 'Plain Bagel',
            category: 'bread',
            subCategory: 'bagels',
            quantity: 3,
            dairy: null,
            flavor: null,
            sugar: false,
            size: 'md',
            ice: 'normal',
            creamCheese: null,
            butter: null,
            bottled: false,
            soda: false,
            menuItemImage: 'plainBagel.png',
          }}
        />,
      )
      .toJSON();
    expect(result).toMatchSnapshot();
  });

  it('renders MyBagItemCreamCheeseText correctly (cream-cheese-text)', async () => {
    render(
      <MyBagItemCreamCheeseText
        myBagItem={{
          orderID: '1',
          menuItemID: 'plainBagel',
          name: 'Plain Bagel',
          category: 'bread',
          subCategory: 'bagels',
          quantity: 3,
          dairy: null,
          flavor: null,
          sugar: false,
          size: 'md',
          ice: 'normal',
          creamCheese: 'Regular',
          butter: null,
          bottled: false,
          soda: false,
          menuItemImage: 'plainBagel.png',
        }}
      />,
    );

    const menuItemButterText = await screen.findAllByTestId('cream-cheese-text');

    expect(menuItemButterText[0]).toHaveClass('MuiTypography-root');
    expect(menuItemButterText[0]).toHaveTextContent('REGULAR CREAM CHEESE');
  });
});
