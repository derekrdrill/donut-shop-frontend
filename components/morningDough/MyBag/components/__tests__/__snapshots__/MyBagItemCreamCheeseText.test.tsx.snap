// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBagItemCreamCheeseText unit tests renders as expected 1`] = `null`;
